openerp/addons/base/ir/ir_qweb/qweb.py

```diff
@@ -0,0 +1,1372 @@
+        expression = el.attrib.pop('t-field')
+        field_options = el.attrib.pop('t-field-options', None)
+        record, field_name = expression.rsplit('.', 1)
+        default_content = self._make_name('default_content')
+
+        content = [
+            # record
+            ast.Assign(
+                targets=[ast.Name(id='record', ctx=ast.Store())],
+                value=self._compile_expr(record)
+            )
+        ]
+
+        body = self._compile_directive_content(el, options)
+        if body:
+            content.extend([
+                # default_content = []
+                ast.Assign(
+                    targets=[ast.Name(id=default_content, ctx=ast.Store())],
+                    value=ast.List(elts=[], ctx=ast.Load())
+                ),
+                # body_call_content(self, default_content.append, values, options)
+                ast.Expr(self._call_body(options,
+                    body=body,
+                    append=ast.Attribute(
+                        value=ast.Name(id=default_content, ctx=ast.Load()),
+                        attr='append',
+                        ctx=ast.Load()
+                    ),
+                    prefix='body_call_content',
+                    lineno=el.sourceline
+                )),
+                # default_content = u''.join(default_content)
+                ast.Assign(
+                    targets=[ast.Name(id=default_content, ctx=ast.Store())],
+                    value=ast.Call(
+                        func=ast.Attribute(
+                            value=ast.Str(u''),
+                            attr='join',
+                            ctx=ast.Load()
+                        ),
+                        args=[
+                            ast.Name(id=default_content, ctx=ast.Load())
+                        ],
+                        keywords=[], starargs=None, kwargs=None
+                    )
+                ),
+            ])
+        else:
+            # default_content = u''
+            content.append(ast.Assign(
+                targets=[ast.Name(id=default_content, ctx=ast.Store())],
+                value=ast.Str(u'')
+            ))
+
+        content.extend([
+            # t_attrs, content = self._get_field(record, field_name, expression, field options, template options, values)
+            ast.Assign(
+                targets=[ast.Tuple(elts=[ast.Name(id='t_attrs', ctx=ast.Store()), ast.Name(id='content', ctx=ast.Store())], ctx=ast.Store())],
+                value=ast.Call(
+                    func=ast.Attribute(
+                        value=ast.Name(id='self', ctx=ast.Load()),
+                        attr='_get_field',
+                        ctx=ast.Load()
+                    ),
+                    args=[
+                        ast.Name(id='record', ctx=ast.Load()),
+                        ast.Str(field_name),
+                        ast.Str(expression),
+                        ast.Name(id=default_content, ctx=ast.Load()),
```

Comment:
I find this a bit problematic: the default content is *always* evaluated, even if the field value is not null.
I would evaluate it only when necessary.